### ISSUE TITLE
Deno.serve handle leaked in test; test fails

### ISSUE THREAD

**Issue by mlandalv** [denoland/deno#22378]
Version: Deno 1.40.4

I think this is a bug but it could also be me doing some rookie mistake.

When using Deno.serve in tests I can't get the test to exit gracefully.

**Reproduce**
1. Create a HttpServer with `Deno.serve`, pass a signal used to exit.
2. In the `onListen` handler add a few test steps.
3. When all steps are done abort the signal so the server finishes.
4. Shutdown the server.

**Expected**
I expect the tests steps to pass and the test to exit gracefully.

**Result**
Step 1 fails
Step 2 passes
Test fails with `error: Leaking async ops`

(The test steps are identical so a bit weird that they have different outcome.)

**Reproduce**
Very simple example to reproduce.

```ts
// test.ts
import { assertEquals } from 'https://deno.land/std@0.215.0/assert/mod.ts';

Deno.test('Test example', async (t) => {
    const ac = new AbortController();

    const server = Deno.serve({
        async onListen(e)  {
            console.log(`1. Devserver started on ${e.hostname}:${e.port}`);
            const baseUrl = new URL(`http://${e.hostname}:${e.port}`);

            await t.step('step 1', async () => {
                console.log('2. Step 1 start');
                const res = await fetch(new URL('/', baseUrl));
    
                assertEquals(res.status, 200);
                assertEquals(await res.text(), 'text');

                console.log('3. Step 1 done');
            });

            await t.step('step 2', async () => {
                console.log('4. Step 2 start');
                const res = await fetch(new URL('/', baseUrl));
    
                assertEquals(res.status, 200);
                assertEquals(await res.text(), 'text');

                console.log('5. Step 2 done');
            });

            console.log('6. All steps done, ac.abort()');
            ac.abort();
        },
        signal: ac.signal,
    }, (_req) => {
        return new Response('text');
    });

    await server.finished;
    console.log('7. Devserver finished');

    // "Gracefully close the server."
    await server.shutdown();
    console.log('8. Cleanup done');
});
```

```bash
> deno test -A --trace-ops test.ts
running 1 test from ./test.ts
Test example ...
------- output -------
1. Devserver started on localhost:8000
----- output end -----
  step 1 ...
------- output -------
2. Step 1 start
3. Step 1 done
----- output end -----
  step 1 ... FAILED (5ms)
  step 2 ...
------- output -------
4. Step 2 start
5. Step 2 done
----- output end -----
  step 2 ... ok (0ms)
------- output -------
6. All steps done, ac.abort()
7. Devserver finished
8. Cleanup done
----- output end -----
Test example ... FAILED (due to 1 failed step) (7ms)

 ERRORS 

Test example ... step 1 => ./test.ts:11:21
error: Leaking async ops:
  - 1 async operation to op_http_wait was started in this test, but never completed. The operation was started here:
    at handleOpCallTracing (ext:core/00_infra.js:137:42)
    at op_http_wait (ext:core/00_infra.js:292:21)
    at ext:deno_http/00_serve.js:671:26
    at eventLoopTick (ext:core/01_core.js:65:7)
    at async file:///tmp/test.ts:39:5
    at async innerWrapped (ext:cli/40_test.js:358:5)
    at async asyncOpSanitizer (ext:cli/40_test.js:180:27)
    at async exitSanitizer (ext:cli/40_test.js:287:27)
    at async outerWrapped (ext:cli/40_test.js:301:14)

 FAILURES 

Test example ... step 1 => ./test.ts:11:21

FAILED | 0 passed (1 step) | 1 failed (1 step) (8ms)

error: Test failed
```

**Comment by mmastrac** [comment]
If you remove the test steps, does it pass? This might get fixed with the new op sanitizer.

**Comment by mlandalv** [comment]
Yes, it passes it the steps are removed.

**Comment by acrodrig** [comment]
Having same issue. Was the new op sanitizer launched?

**Comment by mmastrac** [comment]
This passes in the soon-to-be-released 1.41.1 for me with the new op sanitizer. Closing for now.
